### ISSUE TITLE
Synapse role assignment failure (AzureUSGovernment)

### ISSUE THREAD

**Issue by lucashby** [hashicorp/terraform-provider-azurerm#15443]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v1.1.2
azurerm v2.96.0

### Affected Resource(s)

* `azurerm_synapse_role_assignment`

### Terraform Configuration Files

```hcl
resource "azurerm_synapse_role_assignment" "tf_synapse_role_assignment" {
  depends_on           = [time_sleep.wait_60_seconds]
  synapse_workspace_id = azurerm_synapse_workspace.synapse_workspace.id
  role_name            = "Synapse Administrator"
  principal_id         = var.aad_group_id
}
```

### Debug Output

[TFE Synapse Error Debug](https://gist.github.com/lucashby/63e7d1d592ae71fa5903f8be84c47795)

### Panic Output

2022-02-15T18:31:49.399Z [ERROR] vertex "azurerm_synapse_role_assignment.tf_synapse_role_assignment" error: Synapse is not supported in this Azure Environment

### Expected Behaviour

Synapse role assignment should have been created.

### Actual Behaviour

Received error stating Synapse is not available in this Azure environment.
We are using Synapse already in this Azure environment so the error is invalid.

### Steps to Reproduce

1. Attempt to tf plan/apply to create an azurerm_synapse_role_assignment for a workspace in AzureUSGovernment cloud

### Important Factoids

Running in AzureUSGovernment

**Comment by lucashby** [comment]
I believe this is also going to impact many of the other resources that begin with "azurerm_synapse_...", such as azurerm_synapse_managed_private_endpoint, which also refuses to work, but certainly should.

**Comment by tombuildsstuff** [comment]
This'll need updating upstream: https://github.com/Azure/go-autorest/blob/b3899c1057425994796c92293e931f334af63b4e/autorest/azure/environments.go#L167

**Comment by lucashby** [comment]
> This'll need updating upstream: https://github.com/Azure/go-autorest/blob/b3899c1057425994796c92293e931f334af63b4e/autorest/azure/environments.go#L167

How do we get that update requested?

**Comment by lucashby** [comment]
> > This'll need updating upstream: https://github.com/Azure/go-autorest/blob/b3899c1057425994796c92293e931f334af63b4e/autorest/azure/environments.go#L167
> 
> How do we get that update requested?

I have submitted a GH issue to the Azure autorest repo:  https://github.com/Azure/go-autorest/issues/687

**Comment by lucashby** [comment]
> This'll need updating upstream: https://github.com/Azure/go-autorest/blob/b3899c1057425994796c92293e931f334af63b4e/autorest/azure/environments.go#L167

According to the autorest issue I submitted, the issue was closed with the comment:

"Fixed in autorest@v0.11.26"
